apps/address-book/contracts/AddressBook.sol

```diff
@@ -1,86 +1,104 @@
-contract AddressBook is AragonApp {
+    /// Error string constants
+    string private constant ERROR_NOT_FOUND = "ENTRY_DOES_NOT_EXIST";
+    string private constant ERROR_EXISTS = "ENTRY_ALREADY_EXISTS";
+    string private constant ERROR_CID_MALFORMED = "CID_MALFORMED";
 
-    // The entries in the registry.
+    /// The entries in the registry
     mapping(address => string) entries;
-    // Fired when an entry is added to the registry.
-    event EntryAdded(address addr);
-    // Fired when an entry is removed from the registry.
-    event EntryRemoved(address addr);
 
-    bytes32 public constant ADD_ENTRY_ROLE = keccak256("ADD_ENTRY_ROLE");
-    bytes32 public constant REMOVE_ENTRY_ROLE = keccak256("REMOVE_ENTRY_ROLE");
+    /// Events
+    event EntryAdded(address addr); /// Fired when an entry is added to the registry
+    event EntryRemoved(address addr); /// Fired when an entry is removed from the registry
+    event EntryUpdated(address addr); /// Fired when an entry is updated with a new CID.
+
+    /**
+     * @dev Guard to check existence of address in the registry
+     * @param _addr The address to enforce its existence in the registry
+     */
+    modifier entryExists(address _addr) {
+        require(bytes(entries[_addr]).length != 0, ERROR_NOT_FOUND);
+        _;
+    }
 
+    /**
+     * @notice Initialize AddressBook app`
+     * @dev Initializes the app, this is the Aragon custom constructor
+     */
     function initialize() external onlyInit {
         initialized();
     }
 
     /**
-     * @notice Add the address `_addr` to the registry.
+     * @notice Add the entity `_cid` with address `_addr` to the registry.
      * @param _addr The address of the entry to add to the registry
      * @param _cid The IPFS hash of the entry to add to the registry
      */
-    function addEntry(
-        address _addr,
-        string _cid
-    ) public auth(ADD_ENTRY_ROLE)
-    {
-        require(bytes(entries[_addr]).length == 0, "entry exists with that address");
-        require(bytes(_cid).length == 46, "CID malformed");
+    function addEntry(address _addr, string _cid) public auth(ADD_ENTRY_ROLE) {
+        require(bytes(entries[_addr]).length == 0, ERROR_EXISTS);
+        require(bytes(_cid).length == 46, ERROR_CID_MALFORMED);
 
         entries[_addr] = _cid;
-
         emit EntryAdded(_addr);
     }
 
     /**
-     * @notice Remove address `_addr` from the registry.
+     * @notice Remove entity `_cid` with address `_addr` from the registry.
      * @param _addr The ID of the entry to remove
+     * @param _cid The IPFS hash of the entry to remove from the registry; used only for radpec here
      */
-    function removeEntry(
-        address _addr
-    ) public auth(REMOVE_ENTRY_ROLE)
-    {
-        require(bytes(entries[_addr]).length != 0, "entry does not exist");
-
+    function removeEntry(address _addr, string _cid) public auth(REMOVE_ENTRY_ROLE) entryExists(_addr) {
```

Comment:
If we're going to reference the CID in the radspec, and we expect it to be meaningful we need to require the CID matches the current CID for this entry.